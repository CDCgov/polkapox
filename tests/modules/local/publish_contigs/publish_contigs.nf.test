nextflow_process {

    name "Test Process PUBLISH_CONTIGS"
    script "modules/local/publish_contigs/publish_contigs.nf"
    process "PUBLISH_CONTIGS"

    test("Publish Contig Test") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_sub', single_end: true ],
                    file('${projectDir}/tests/nf-test-data/test_sub.final.fa', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
