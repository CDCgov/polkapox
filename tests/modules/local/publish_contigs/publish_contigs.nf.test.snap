{
    "Publish Contig Test": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "versions.yml:md5,887df3ebc0dceef20bd08e960a6db4f3"
                ],
                "versions": [
                    "versions.yml:md5,887df3ebc0dceef20bd08e960a6db4f3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-07T16:50:05.599985693"
    }
}